client/lib/feed-lookup/index.js

```diff
@@ -0,0 +1,52 @@
+// External Dependencies
+var isEmpty = require( 'lodash/isEmpty' ),
+	head = require( 'lodash/head' ),
+	lruCache = require( 'lru-cache' );
+
+// Internal Dependencies
+var inflight = require( 'lib/inflight' ),
+	wpcom = require( 'lib/wp' );
+
+const cache = lruCache( 10 );
+
+function requestKey( feedId ) {
+	return `feed-lookup-${feedId}`;
+}
+
+function discover( feedUrl ) {
+	const key = requestKey( feedUrl );
+
+	if ( inflight.requestInflight( key ) ) {
+		return cache.get( feedUrl );
+	}
+
+	return wpcom.undocumented().discoverFeed(
+		{ url: feedUrl }
+	);
+};
+
+function feedLookup( feedUrl ) {
+	var feedId = cache.get( feedUrl );
```

Comment:
can we rename this to something like `promiseForFeedId`? It's not the actual feed ID and I think this will confuse folks who walk into this code later.